httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/HttpRFC7578Multipart.java

```diff
@@ -101,8 +101,12 @@ protected void formatMultipartHeader(final MultipartPart part, final OutputStrea
                     writeBytes("=\"", out);
                     if (value != null) {
                         if (name.equalsIgnoreCase(MimeConsts.FIELD_PARAM_FILENAME_START)) {
-                            final String encodedValue = "UTF-8''" + PercentCodec.RFC5987.encode(value);
-                            writeBytes(encodedValue, StandardCharsets.US_ASCII, out);
+                            if (value.startsWith("UTF-8''")) {
+                                writeBytes(value, StandardCharsets.US_ASCII, out);
+                            } else {
+                                final String encodedValue = "UTF-8''" + PercentCodec.RFC5987.encode(value);
```

Comment:
@arturobernalg Could you please use `StringBuilder` and `PercentCodec#encode(StringBuilder, CharSequence)` here? Small optimization but still a bit less garbage.

Reply:
@ok2c  done